src/crc32.rs

```diff
@@ -21,6 +21,14 @@ impl Crc32 {
     pub fn new(crc: CRC) -> Self {
         let mut new = Self { periph: crc };
         new.init();
+
+        unsafe {
+            // NOTE(unsafe) this reference will only be used for atomic writes with no side effects.
+            let rcc_raw = unsafe { &(*RCC::ptr()) };
+            // Enable CRC clock
+            rcc_raw.ahb1enr.modify(|_, w| w.crcen().enabled());
```

Comment:
Please use bitbanding to change the flag instead, cf. `gpio.rs`.

Also after enabling the clocks, you might want to throw in a `cortex_m::asm::dsb();` for good measure to work around an erratum, also cf. `gpio.rs`.

Reply:
Done.